src/main/scala/org/bitcoins/core/protocol/transaction/TransactionOutPoint.scala

```diff
@@ -21,36 +22,42 @@ sealed trait TransactionOutPoint extends NetworkElement {
  *
     * @return
     */
-  def vout : Int
+  def vout : UInt32
 
   //https://bitcoin.org/en/developer-reference#outpoint
   override def size = 36
 
   override def hex = RawTransactionOutPointParser.write(this)
 }
 
+/**
+  * UInt32s cannot hold negative numbers, but sometimes the Bitcoin Protocol requires the vout to be -1, which is serialized
+  * as "0xFFFFFFFF".
+  * https://github.com/bitcoin/bitcoin/blob/d612837814020ae832499d18e6ee5eb919a87907/src/primitives/transaction.h
+  * http://stackoverflow.com/questions/2711522/what-happens-if-i-assign-a-negative-value-to-an-unsigned-variable
+  */
 case object EmptyTransactionOutPoint extends TransactionOutPoint {
   def txId = DoubleSha256Digest(
     BitcoinSUtil.decodeHex("0000000000000000000000000000000000000000000000000000000000000000"))
-  def vout = -1
+  def vout =  UInt32("ffffffff")
 }
 
 object TransactionOutPoint extends Factory[TransactionOutPoint] {
 
-  private sealed case class TransactionOutPointImpl(txId : DoubleSha256Digest, vout : Int) extends TransactionOutPoint
+  private sealed case class TransactionOutPointImpl(txId : DoubleSha256Digest, vout : UInt32) extends TransactionOutPoint
   /**
     * Creates a transaction outpoint from a TransactionOutput & it's parent transaction
  *
     * @param output
     * @return
     */
   private def factory(output : TransactionOutput, parentTransaction : Transaction) : TransactionOutPoint = {
-    val indexOfOutput = parentTransaction.outputs.indexOf(output)
-    if (indexOfOutput == -1) throw new RuntimeException("This output is not contained in the parent transaction")
+    val indexOfOutput = UInt32(parentTransaction.outputs.indexOf(output))
+    if (indexOfOutput.underlying.toInt == (-1)) throw new RuntimeException("This output is not contained in the parent transaction")
```

Comment:
This can be just `.toInt` now instead of `underlying.toInt`, also the parentheses around `-1` are unnecessary.